e2e/e2e_node_drop_test.go

```diff
@@ -10,8 +13,57 @@ func TestE2E_NodeDrop(t *testing.T) {
 	c.Start()
 
 	// wait for two heights and stop node 1
-	c.WaitForHeight(2, 1*time.Minute)
+	c.WaitForHeight(2, 1*time.Minute, false)
 
 	c.StopNode("ptr_0")
-	c.WaitForHeight(15, 1*time.Minute, []string{"ptr_1", "ptr_2", "ptr_3", "ptr_4"})
+	c.WaitForHeight(15, 1*time.Minute, false, generateNodeNames(1, 4, "ptr_"))
+}
+
+func TestE2E_NetworkChurn(t *testing.T) {
+	rand.Seed(time.Now().Unix())
+	nodeCount := 20
+	maxFaulty := nodeCount/3 - 1
+	const prefix = "ptr_"
+	c := newPBFTCluster(t, "network_churn", "ptr", nodeCount)
+	c.Start()
+	runningNodeCount := nodeCount
+	// randomly stop nodes every 3 seconds
+	executeInTimerAndWait(3*time.Second, 30*time.Second, func(_ time.Duration) {
```

Comment:
Do not make this an E2E test but a fuzz test. Create a different file called ```fuzz_network_churn_test.go``` and call the test ```TestFuzz_**``` instead.

Reply:
:+1:

Reply:
Yes, everything that is not "deterministic" and has some random scenarios in the test itself should not be part of e2e_, all of those go to fuzz_xxx_test.go but keep on the same package.